vmware-base/src/com/cloud/hypervisor/vmware/mo/VirtualMachineMO.java

```diff
@@ -2137,7 +2141,8 @@ public int getScsiDiskControllerKey(String diskController) throws Exception {
         }
 
         assert (false);
-        throw new Exception(diskController + " Controller Not Found");
+        throw new Exception("Scsi disk controller of type " + diskController + " not found among configured devices:" + deviceList +
+                ". Unable to find and return scsi disk controller key.");
```

Comment:
Throwing ``Exception`` is an anti-pattern.  Use a specific checked or unchecked exception type to better express the nature of the problem (e.g. ``IllegalStateException``).  Also, since an exception is being thrown, ``assert(false)`` is not necessary and can be removed.